In [55]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

olist_customer_df = pd.read_csv("clean_data/cleaned_customers.csv")
olist_sellers_df=pd.read_csv("clean_data/cleaned_sellers.csv")
olist_products_df=pd.read_csv("clean_data/cleaned_products.csv")
olist_geolocation_df = pd.read_csv("clean_data/cleaned_new_geolocation.csv")
olist_orders_df=pd.read_csv("clean_data/cleaned_orders.csv")
olist_order_items_df=pd.read_csv("clean_data/cleaned_orders_items.csv")
olist_order_payments_df =pd.read_csv("clean_data/cleaned_orders_payments.csv")
olist_order_review_df  =pd.read_csv("clean_data/cleaned_orders_reviews.csv")

In [54]:
olist_order_review_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 10 columns):
 #   Column                     Non-Null Count   Dtype 
---  ------                     --------------   ----- 
 0   review_id                  100000 non-null  object
 1   order_id                   100000 non-null  object
 2   review_score               100000 non-null  int64 
 3   review_comment_title       11715 non-null   object
 4   review_comment_message     41753 non-null   object
 5   review_creation_date       100000 non-null  object
 6   review_answer_timestamp    100000 non-null  object
 7   cleaned_review_comment     41735 non-null   object
 8   translated_review_comment  41601 non-null   object
 9   translated_comment_title   11624 non-null   object
dtypes: int64(1), object(9)
memory usage: 7.6+ MB


In [32]:
olist_products_df[olist_products_df['product_category_name']=='unclassified']
# olist_products_df[olist_products_df['product_category_name'].isna()]

product_id product_category_name  \
105    a41e356c76fab66334f36de622ecbd3a          unclassified   
128    d8dee61c2034d6d075997acef1870e9b          unclassified   
145    56139431d72cd51f19eb9f7dae4d1617          unclassified   
154    46b48281eb6d663ced748f324108c733          unclassified   
197    5fb61f482620cb672f5e586bb132eae9          unclassified   
...                                 ...                   ...   
32515  b0a0c5dd78e644373b199380612c350a          unclassified   
32589  10dbe0fbaa2c505123c17fdc34a63c56          unclassified   
32616  bd2ada37b58ae94cc838b9c0569fecd8          unclassified   
32772  fa51e914046aab32764c41356b9d4ea4          unclassified   
32852  c4ceee876c82b8328e9c293fa0e1989b          unclassified   

      product_category_name_translated  product_name_lenght  \
105                                NaN                  NaN   
128                                NaN                  NaN   
145                                NaN                  NaN   
154                                NaN                  NaN   
197                                NaN                  NaN   
...                                ...                  ...   
32515                              NaN                  NaN   
32589                              NaN                  NaN   
32616                              NaN                  NaN   
32772                              NaN                  NaN   
32852                              NaN                  NaN   

       product_description_lenght  product_photos_qty  product_weight_g  \
105                           NaN                 NaN             650.0   
128                           NaN                 NaN             300.0   
145                           NaN                 NaN             200.0   
154                           NaN                 NaN           18500.0   
197                           NaN                 NaN             300.0   
...                           ...                 ...               ...   
32515                         NaN                 NaN            1800.0   
32589                         NaN                 NaN             800.0   
32616                         NaN                 NaN             200.0   
32772                         NaN                 NaN            1300.0   
32852                         NaN                 NaN             700.0   

       product_length_cm  product_height_cm  product_width_cm  
105                 17.0               14.0              12.0  
128                 16.0                7.0              20.0  
145                 20.0               20.0              20.0  
154                 41.0               30.0              41.0  
197                 35.0                7.0              12.0  
...                  ...                ...               ...  
32515               30.0               20.0              70.0  
32589               30.0               10.0              23.0  
32616               21.0                8.0              16.0  
32772               45.0               16.0              45.0  
32852               28.0                3.0              43.0  

[610 rows x 10 columns]

#### **olist_customer_df**
Description: Details of olist's customers
Columns: [5] customer_id, customer_unique_id, customer_zip_code_prefix, customer_city, customer_state 
Rows: 99441 no null values

#### **olist_sellers_df**
Description: Details of olist sellers
Columns: [4] seller_id, seller_zip_code_prefix, seller_city, seller_state
Rows: 3095  no null values

#### **olist_products_df**
Description: Details of products available on Olist ecommerce platform
Columns: [9] product_id, product_category_name, product_category_name_translated, product_name_lenght, product_description_lenght, product_photos_qty, product_weight_g, product_length_cm, product_height_cm, product_width_cm
Rows: 32951, 610 products are unclassified

#### **olist_geolocation_df**
Description: Geographical location information
Columns: [5] geolocation_zip_code_prefix, geolocation_state, geolocation_city, geolocation_lat, geolocation_lng 
Rows: 19016 unique zipcode

#### **olist_orders_df**
Description: The main order dataset 
Columns: [8] order_id, customer_id, order_status, order_purchase_timestamp, order_approved_at, order_delivered_carrier_date,order_delivered_customer_date, order_estimated_delivery_date         
Rows: 99441 rows

#### **olist_order_items_df**
Description: comprehensive view of individual items within each order
Columns: [9] id_pk, order_id, order_item_id, product_id, seller_id, shipping_limit_date, price, freight_value, total_freight_cost
Rows: 112650 rows

#### **olist_order_payments_df**
Description: Details on payments, transactions
Columns:[6] id_pk, order_id, payment_sequential, payment_type, payment_installments, payment_value
Rows: 103883 Rows

#### **olist_order_payments_df**
Description: Customer reviews and ratings
Columns:[10] review_id, order_id, review_score, review_comment_title, review_comment_message, review_creation_date, review_answer_timestamp, cleaned_review_comment, translated_review_comment, ranslated_comment_title 
Rows: 100000  Rows